flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ShowProceduresOperation.java

```diff
@@ -65,7 +65,7 @@ protected Collection<String> retrieveDataForTableResult(Context ctx) {
             if (preposition == null) {
                 // it's to show current_catalog.current_database
                 return catalogManager
-                        .getCatalogOrError(catalogManager.getCurrentCatalog())
+                        .getCatalogOrThrowException(catalogManager.getCurrentCatalog())
```

Comment:
there should be no call in this class to `catalogManager.getCurrentCatalog()`. Maybe I'm missing some context?

Reply:
here it is a case with `preposition == null`
i.e. when no catalog or database was specified e.g. 
```sql
SHOW PROCEDURES
```
in this case we need to use current catalog and database.

In theory we could precalculate it on converters level and then here use only `catalogName`, `databaseName`

Reply:
updated, now there is no `getCurrentCatalog`/`getCurrentDatabase` calls in operations